### PR TITLE
add joe-avax-more strat and pangolin superfarms

### PR DESCRIPTION
**Pull request by zeroecksYeti**
Also reorder slots and controllers in helpers.ts

### CODE REVIEW

File: contracts/strategies/pangolin/strategy-png-avax-ust.sol [hunk 1]
```diff
@@ -0,0 +1,156 @@
+pragma solidity ^0.6.7;
+
+import "../strategy-png-minichef-farm-base.sol";
+
+contract StrategyPngAvaxUst is StrategyPngMiniChefFarmBase {
+    uint256 public _poolId = 75;
```
Comment (zeroecksYeti) [inline contracts/strategies/pangolin/strategy-png-avax-ust.sol:6]: Placeholder `_poolId`. Needs to be updated

File: contracts/strategies/pangolin/strategy-png-usdc-ust.sol [hunk 2]
```diff
@@ -0,0 +1,171 @@
+pragma solidity ^0.6.7;
+
+import "../strategy-png-minichef-farm-base.sol";
+
+contract StrategyPngUsdcUst is StrategyPngMiniChefFarmBase {
+    uint256 public _poolId = 75;
```
Comment (zeroecksYeti) [inline contracts/strategies/pangolin/strategy-png-usdc-ust.sol:6]: Also placeholder `_poolId`

File: contracts/strategies/pangolin/strategy-png-usdc-ust.sol [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+            uint256 _keep2 = _ust.mul(keep).div(keepMax);
+            if (_keep2 > 0){
+                _takeFeeUstToSnob(_keep2);
+            }
+            
+            _ust = IERC20(ust).balanceOf(address(this));
+        }
+
+        // In the case of PNG Rewards, swap PNG for USDC and UST
+        _png = IERC20(png).balanceOf(address(this));
```
Comment (zeroecksYeti) [inline contracts/strategies/pangolin/strategy-png-usdc-ust.sol:109]: I've already checked the `_png` balanceof on line 87...unsure if this should be removed, or if it costs little enough to leave in.

File: contracts/strategies/pangolin/strategy-png-avax-ust.sol [hunk 4]
```diff
@@ -80,18 +80,11 @@ contract StrategyPngAvaxUst is StrategyPngMiniChefFarmBase {
...
-            if (_keep > 0) {
-                _takeFeePngToSnob(_keep);
-            }
-
-            _png = IERC20(png).balanceOf(address(this));  
-        }
-
+        
         if (_ust > 0) {
             uint256 _keep2 = _ust.mul(keep).div(keepMax);
```
Comment (Auroter) [inline contracts/strategies/pangolin/strategy-png-avax-ust.sol:89]: It looks like we are setting the 'keep' value right before we use it each time (_takeFeeXYZToSnob()) so there is no need to specifically call this 'keep' variable 'keep2'

File: contracts/strategies/pangolin/strategy-png-usdc-ust.sol (hunk 3 above)
Comment (Auroter) [reply contracts/strategies/pangolin/strategy-png-usdc-ust.sol:164]: This is true for wavax and luna as well.  Each time a fee is deducted, the _luna, _png, _wavax variables are updated accordingly.  So it is not needed, but it does make each codeblock easy to read.

**Review by Auroter — APPROVED** [review]

File: contracts/strategies/pangolin/strategy-png-usdc-ust.sol [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+
+        if (_luna > 0) {
+            uint256 _keep = _luna.mul(keep).div(keepMax);
+            if (_keep > 0){
+                _takeFeeLunaToSnob(_keep);
+            }
+            
+            _luna = IERC20(wavax).balanceOf(address(this));
+        }
+
```
Comment (AbigailTCameron) [inline contracts/strategies/pangolin/strategy-png-usdc-ust.sol:153]: if (_ust) then we swap half for usdc
if (_usdc) then we swap half for ust

File: contracts/strategies/traderJoe/strategy-joe-avax-ust.sol [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+            IERC20(wavax).safeApprove(joeRouter, 0);
+            IERC20(wavax).safeApprove(joeRouter, _wavax.div(2));   
+            _swapTraderJoe(wavax, luna, _wavax.div(2)); 
+        }
+
+        // In the case of UST Rewards, swap half UST for WAVAX
+        if(_ust > 0){
+            IERC20(ust).safeApprove(joeRouter, 0);
+            IERC20(ust).safeApprove(joeRouter, _ust.div(2));   
+            _swapTraderJoe(ust, luna, _ust.div(2));
```
Comment (AbigailTCameron) [inline contracts/strategies/traderJoe/strategy-joe-avax-ust.sol:134]: _swapTraderJoe(ust, **wavax**, _ust.div(2));

File: contracts/strategies/traderJoe/strategy-joe-avax-ust.sol [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+            }
+
+            _joe = IERC20(joe).balanceOf(address(this));
+        }
+
+        // In the case of AVAX Rewards, swap half WAVAX for LUNA
+        if(_wavax > 0){
+            IERC20(wavax).safeApprove(joeRouter, 0);
+            IERC20(wavax).safeApprove(joeRouter, _wavax.div(2));   
+            _swapTraderJoe(wavax, luna, _wavax.div(2));
```
Comment (AbigailTCameron) [inline contracts/strategies/traderJoe/strategy-joe-avax-ust.sol:127]: _swapTraderJoe(wavax, **ust,** _wavax.div(2));

File: contracts/strategies/traderJoe/strategy-joe-avax-ust.sol [hunk 8]
```diff
@@ -0,0 +1,217 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.6.7;
+
+import "../strategy-joe-rush-farm-base.sol";
+
+contract StrategyJoeAvaxUst is StrategyJoeRushFarmBase {
+
+    uint256 public avax_ust_poolId = 46;
```
Comment (AbigailTCameron) [inline contracts/strategies/traderJoe/strategy-joe-avax-ust.sol:8]: 45

**Review by AbigailTCameron — APPROVED** [review]
LGTM!

**Pull request merged**